tests/ubench/event_setup.cpp

```diff
@@ -0,0 +1,207 @@
+// Compare methods for performing the "event-setup" step for mc cell groups.
+// The key concern is how to take an unsorted set of events
+//
+// TODO: We assume that the cells in a cell group are numbered contiguously,
+// i.e. 0:ncells-1. The cells in an mc_cell_group are not typically thus,
+// instead a hash table is used to look up the cell_group local index from the
+// gid. A similar lookup should be added to theses tests, to moare accurately
+// reflect the mc_cell_group implementation.
+//
+// TODO: The staged_events output is a vector of postsynaptic_spike_event, not
+// a deliverable event.
+
+#include <random>
+#include <vector>
+
+#include <event_binner.hpp>
+#include <event_queue.hpp>
+#include <backends/event.hpp>
+
+#include <benchmark/benchmark.h>
+
+using namespace nest::mc;
+
+std::vector<postsynaptic_spike_event> generate_inputs(size_t ncells, size_t ev_per_cell) {
+    std::vector<postsynaptic_spike_event> input_events;
+    std::default_random_engine engine;
+    std::uniform_int_distribution<cell_gid_type>(0u, ncells);
+
+    std::default_random_engine gen;
+    std::uniform_int_distribution<cell_gid_type>
+        gid_dist(0u, ncells-1);
+    std::uniform_real_distribution<float>
+        time_dist(0.f, 1.f);
+
+    input_events.reserve(ncells*ev_per_cell);
+    for (std::size_t i=0; i<ncells*ev_per_cell; ++i) {
+        postsynaptic_spike_event ev;
+        auto gid = gid_dist(gen);
+        auto t = time_dist(gen);
+        ev.target = {cell_gid_type(gid), cell_lid_type(0)};
+        ev.time = t;
+        ev.weight = 0;
+        input_events.push_back(ev);
+    }
+
+    return input_events;
+}
+
+void single_queue(benchmark::State& state) {
+    using pev = postsynaptic_spike_event;
+
+    const std::size_t ncells = state.range(0);
+    const std::size_t ev_per_cell = state.range(1);
+
+    // state
+    std::vector<pev> input_events = generate_inputs(ncells, ev_per_cell);
+
+    //auto binner = event_binner(binning_kind::regular, 0.001);
```

Comment:
We should probably nuke the binner in this benchmark entirely — still not clear if we should even use it by default, given we now have much lower per event-time overhead of delivery.

Reply:
Done.